src/validation.cpp

```diff
@@ -1294,15 +1295,23 @@ void CChainState::InitCoinsCache(size_t cache_size_bytes)
 // `const` so that `CValidationInterface` clients (which are given a `const CChainState*`)
 // can call it.
 //
+// The file .ibd_complete is used to latch the datadir to IBD false across restarts
+//
 bool CChainState::IsInitialBlockDownload() const
 {
+    static const fs::path latch_filename = GetDataDir() / ".ibd_complete";
     // Optimization: pre-test latch before taking the lock.
     if (m_cached_finished_ibd.load(std::memory_order_relaxed))
         return false;
 
     LOCK(cs_main);
     if (m_cached_finished_ibd.load(std::memory_order_relaxed))
         return false;
+    if (fs::exists(latch_filename)) {
+        LogPrintf("Leaving InitialBlockDownload (latching to false)\n");
+        m_cached_finished_ibd.store(true, std::memory_order_relaxed);
+        return false;
```

Comment:
This will set ibd to false, when `-reindex`? If yes, that seems wrong

Reply:
yes, it is wrong